src/cmap/stream_description.ts

```diff
@@ -58,6 +59,9 @@ export class StreamDescription {
       return;
     }
     this.type = parseServerType(response);
+    if (response.connectionId) {
+      this.serverConnectionId = this.parseServerConnectionID(response.connectionId);
```

Comment:
Could we consider always defining the property and setting it to null by default? 

I don't think `connectionId` would ever be 0 but if it were this code would ignore it because it is falsey, just always assigning and defaulting to null removes the need to check for that.

Ref. https://jira.mongodb.org/browse/NODE-4068

Reply:
What was the behavior before server v4.2 / did it have a connectionId back then? I feel it's much safer to check if a value exists in the `Document`, especially since it could contain anything. Currently, the parseServerConnectionId method cannot handle a nullish value and would throw an error.

Reply:
> What was the behavior before server v4.2 / did it have a connectionId back then?

There was no connectionId property on the hello response, so we can check for the presence of the key: `'connectionId' in response` or `response.connectionId == null` and set a default of null

Reply:
Sounds good, just made these changes.